app/models/master_file.rb

```diff
@@ -605,9 +606,11 @@ def extract_frame(options={})
     end
 
     frame_size = (options[:size].nil? or options[:size] == 'auto') ? self.original_frame_size : options[:size]
+    # Handle existing files that may have inaccurate display_aspect_ratio saved.
+    aspect_ratio = actual_aspect_ratio != self.display_aspect_ratio.to_f ? actual_aspect_ratio : self.display_aspect_ratio
```

Comment:
Since this seems like an edge case I wonder if a backfill might be better than trying to handle the case dynamically.  After fixing `display_aspect_ratio` the backfill could regenerate the poster and thumbnail.

Reply:
Are you thinking another migration rake task for this or would a different approach be better? And, regardless of approach, should we make an issue specifically for creating the backfill or just wrap it up in this work?

Reply:
I think just having instructions in the upgrade notes is reasonable. You're right that this situation is, in general, pretty edge case so I'm fine with conserving the effort.